hw02-generics/src/main/java/homework/CustomerService.java

```diff
@@ -0,0 +1,38 @@
+package homework;
+
+import java.util.Comparator;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class CustomerService {
+
+    private final TreeMap<Customer, String> innerMap;
```

Comment:
```suggestion
    private final NavigableMap<Customer, String> innerMap;
```
Тип лучше объявлять минимально необходимым интерфейсом.

Reply:
поправил